### PR TITLE
type-connect part 2: constants, actions, engine

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] New protocol generator
- [x] Don't ignore most of node_modules. See if the new flow handles caching better now (should be)
- [x] Incoming calls (rpcSaga etc) no longer pass state in. If you need state you can do Saga.call() to get access
- [x] New typed connect which helps our connectors

### CODE REVIEW

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -12,34 +12,35 @@ const enabledCalls = json5.parse(fs.readFileSync(path.join(__dirname, 'enabled-c
 var projects = {
   chat1: {
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: new protocol generation, makes a js.flow and .js now

File: shared/actions/chat2/index.js [hunk 2]
```diff
@@ -982,22 +984,27 @@ const desktopNotify = (action: Chat2Gen.DesktopNotificationPayload, state: Typed
     !meta.isMuted // ignore muted convos
   ) {
     logger.info('Sending Chat notification')
-    return Saga.put((dispatch: Dispatch) => {
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: an example of us mixing saga plus redux thunk. Use promises instead

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/util/errors.js.flow [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// @flow
+// Proxy so we don't get import loops. We need rpc gen but it also needs us
```
Comment (akalin-keybase) [inline shared/util/errors.js.flow]: `niceError` is the only fn that uses RPCTypes...perhaps move that into its own file instead?

File: shared/actions/chat2/index.js [hunk 4]
```diff
@@ -982,22 +984,27 @@ const desktopNotify = (action: Chat2Gen.DesktopNotificationPayload, state: Typed
...
+            Chat2Gen.createSelectConversation({
+              conversationIDKey,
+              reason: 'desktopNotification',
+            })
+          ),
+          Saga.put(RouteTreeGen.createSwitchTo({path: [chatTab]})),
+          Saga.put(ConfigGen.createShowMain()),
+        ])
+      )
+      .catch(_ => {})
```
Comment (akalin-keybase) [inline shared/actions/chat2/index.js]: why drop errors here?

Comment (chrisnojima) [reply shared/actions/chat2/index.js]: reject is passed to onClose in notifyPopup, so it only happens if you hit close, so we do nothing

Comment (akalin-keybase) [reply shared/actions/chat2/index.js]: Cool. probably worth commenting

**Pull request merged**